### DISCUSSION TITLE
Regex behavior incorrect for 'hints' kitten?

### DISCUSSION THREAD

**Discussion by rexkerr** [kovidgoyal/kitty#4614] [Q&A] [answered]
Before I submit an issue or try to fix this myself I thought it would be good to get clarification to make sure that I'm not just overlooking or misunderstanding something.

I'm trying to use the hints kitten to map a keystroke to open Jira tasks referenced in git history in the browser.  Let's assume that I have two Jira projects, one named PROJ and one named BUGS, and our tickets are in the form of PROJ-#### or BUGS-####.  I wrote the following marker, and it works as expected:

<img width="540" alt="KittyCapturingGroupsBug_marker" src="https://user-images.githubusercontent.com/399845/152254134-0f2930ae-1f84-4adb-bd03-da10ea8bb713.png">

I then tried to make a hints mapping that runs an existing script that converts the task ID into the full URL, but it doesn't match the same thing:

<img width="540" alt="KittyCapturingGroupsBug" src="https://user-images.githubusercontent.com/399845/152254230-bb163592-5d6c-41bc-951f-f2470a5de8e5.png">

As you can see from the bold-face text, it only matches the part in the capture group, not the full regex.  I suspect that the hints kitty is using its own mechanism to perform the regex match different than the build-in regex match used by the marker.

I can work around this, and it does what I want it to do, by expanding the regex and duplicating the `-[0-9]+` portion over and over, but it makes for an uglier expression.

**Comment by rexkerr** [comment]
Shortly after posting I noticed this comment in the documentation for the hints kitten:

> If you specify a numbered group in the regular expression only the group will be matched.

So, I suspect this has something to do with my problem, but I don't know how to work around it.  I've tried nested groups, but that didn't work either.

**Comment by rexkerr** [comment]
...figured it out, reading closer -- it's an implicitly numbered groups.  Using a named group works as expected:

```"(?P<project>PROJ|BUGS)\-[0-9]+"```

**Reply by rexkerr** [reply]
I spoke too soon.  The documentation also says:

> If you specify named groups and a --program then the  program will be passed arguments corresponding to each named group of  the form key=value.

But that's not what I want.  I want the full match to be passed.  If I copy to clipboard, it copies the match, as expected, and that's why I thought that things were working, but the behavior is different when calling `--program=<script>` than when calling `--program=-`, in which case it sends `proj=PROJ` rather than the expected match of `PROJ-1234`.

**Comment by kovidgoyal** [comment, marked as answer]
(?:PROJ|BUGS)\-[0-9]+

**Reply by rexkerr** [reply]
Awesome, thank you!  I didn't think about the non capturing group option, not something that I need very often.
